### PR TITLE
add Button docs to site

### LINKED ISSUE #2198: Button Documentation

### PR DESCRIPTION
**Pull request by feliciackh**
Closes #2198

### CODE REVIEW

File: site/docs/components/button/examples.mdx [hunk 1]
```diff
@@ -12,47 +14,63 @@ data:
...
 
 ### Icon only
 
-Display an icon-only button with no label when you have limited on-screen space and the icon is globally understood.
+Display an icon-only button with no text when you have limited on-screen space, and the icon is easily understood.
+
+Guidance:
+
+- An icon-only button should be self-explanatory, ideally using well known symbols. If the purpose of the button isn’t clear from the icon alone, add a text description.
+- Use [Tooltip](../components/tooltip) on an icon-only button to describe the action such as “Edit this form”, “Print this document”, or “Lock document state”.
```
Comment (feliciackh) [inline site/docs/components/button/examples.mdx:63]:
```suggestion
- Use [Tooltip](../tooltip) on an icon-only button to describe the action such as “Edit this form”, “Print this document”, or “Lock document state”.
```

**Review by lilyvc — APPROVED** [review]

**Review by alycrys — DISMISSED** [review]

File: site/docs/components/button/accessibility.mdx [hunk 2]
```diff
@@ -9,27 +11,35 @@ data:
 
 ### Best practises
```
Comment (lilyvc) [inline site/docs/components/button/accessibility.mdx:12]:
```suggestion
### Best practices
```

**Review by lilyvc — DISMISSED** [review]

File: site/docs/components/button/accessibility.mdx [hunk 3]
```diff
@@ -1,35 +1,45 @@
...
 
-- Pass an `aria-label` to Button that describes its purpose.
+#### Screen reader behavior:
+
+On focus, the screen reader should:
+
+- Identify the button
+- Read out the text,
+- Express the state
+  - In certain situations, the screen reader may need to indicate if the button has a popup for lists or menus.
```
Comment (libertymayc) [inline site/docs/components/button/accessibility.mdx:28]:
```suggestion
- In certain situations, the screen reader may need to indicate if the button has a popup for lists or menus.
```

Comment (feliciackh) [reply site/docs/components/button/accessibility.mdx:28]: the tab was intended as this is a sub-point for express the state

File: site/docs/components/button/usage.mdx [hunk 4]
```diff
@@ -1,22 +1,32 @@
 ---
-title: Button
+# Leave the frontmatter as is
+title:
+  $ref: ./#/title
 layout: DetailComponent
 sidebar:
+  label: Usage
```
Comment (JoanaMMoreira) [inline site/docs/components/button/usage.mdx:7]:
```suggestion
```

**Review by JoanaMMoreira — DISMISSED** [review]

**Review by JoanaMMoreira — DISMISSED** [review]

File: site/docs/components/button/usage.mdx [hunk 5]
```diff
@@ -9,14 +11,21 @@ data:
...
 
-- Keep the text label short. Ideally one to three words and no more than five. If you’re including an icon in the label, use no more than three words.
+- Keep the text description short, ideally one to three words and no more than five. If you’re including an icon with the text, use no more than three words.
+- For icons, use the Salt Figma icon library. If the icon you need isn’t available, follow the steps in our [Creating Salt icons](<https://www.figma.com/file/ibGX0b0CxUcZU3pn6OrqX7/Salt%3A-Icons-(Community)?type=design&node-id=490-74&mode=design>) guide.
 
-- When including icons, use the Salt Figma icon library. If the icon you need isn’t available, follow the steps in our Creating Salt icons guide.
+#### When to use Button
 
-- Consider a Link instead of the Button component if the user is navigating to a new page or window rather than carrying out an action.
+- To allow the user to execute an action, such as “submit”, “merge” or “upload”. Buttons are typically placed in components and contexts such as [Dialog](../dialog), [Cards](../cards) and [Forms](/salt/patterns/forms).
```
Comment (alycrys) [inline site/docs/components/button/usage.mdx:19]:
```suggestion
- To allow the user to execute an action, such as “submit”, “merge” or “upload”. Buttons are typically placed in components and contexts such as [Dialog](../dialog), [Card](../card) and [Form](/salt/patterns/form).
```

Comment (feliciackh) [reply site/docs/components/button/usage.mdx:19]: Will update for card and keep the same for forms (https://www.saltdesignsystem.com/salt/patterns/forms)

Comment (feliciackh) [inline site/docs/components/button/usage.mdx:19]:
```suggestion
- To allow the user to execute an action, such as “submit”, “merge” or “upload”. Buttons are typically placed in components and contexts such as [Dialog](../dialog), [Cards](../card) and [Forms](/salt/patterns/forms).
```

**Review by JoanaMMoreira — DISMISSED** [review]

**Review by alycrys — APPROVED** [review]

File: site/docs/components/button/accessibility.mdx [hunk 6]
```diff
@@ -1,35 +1,45 @@
...
 
-- If focus is above the button, Tab key press moves focus onto the button.
-- If the button is disabled and is set to receive focus, receives focus.
-- If the button has focus, Tab moves focus out of the button, to next focusable component in the tab order.
+- Tab moves focus onto the button.
+- If the button is disabled and not set to receive focus, Tab moves focus to the next focusable component.
+- If the button has focus, Tab moves focus to the next focusable component in the tabbing order.
 
 </KeyboardControl>
 <KeyboardControl keyOrCombos="Tab + Shift">
```
Comment (origami-z) [inline site/docs/components/button/accessibility.mdx:40]: Why this is "Tab + Shift" but most others are "Shift + Tab"?

File: site/docs/components/button/examples.mdx [hunk 7]
```diff
@@ -12,47 +14,63 @@ data:
...
 </LivePreview>
 
 <LivePreview componentName="button" exampleName="Disabled" >
 
 ### Disabled
 
-Indicates a button that the user can’t interact with.
+Use disabled state for a button that the user can’t press.
+
+A button with the prop `disabled=”true”` will suppress all functionality. If you need a disabled button to be focusable, you also need to pass `focusableWhenDisabled="true"`.
```
Comment (origami-z) [inline site/docs/components/button/examples.mdx:73]: `”true”` is not assignable to `disabled`?

File: site/docs/components/button/accessibility.mdx (hunk 6 above)
Comment (feliciackh) [reply site/docs/components/button/accessibility.mdx:40]: updated to align with all docs

File: site/docs/components/button/accessibility.mdx [hunk 8]
```diff
@@ -37,7 +37,7 @@ On focus, the screen reader should:
 - If the button has focus, Tab moves focus to the next focusable component in the tabbing order.
 
 </KeyboardControl>
-<KeyboardControl keyOrCombos="Tab + Shift">
+<KeyboardControl keyOrCombos="Shift + Tab">
 
 - Tab + Shift moves focus out of the component to the previous component in the tab order.
```
Comment (origami-z) [inline site/docs/components/button/accessibility.mdx:42]: "Shift + Tab" here as well

File: site/docs/components/button/accessibility.mdx [hunk 9]
```diff
@@ -1,35 +1,45 @@
...
 
 <KeyboardControls>
 <KeyboardControl keyOrCombos="Tab">
 
-- If focus is above the button, Tab key press moves focus onto the button.
-- If the button is disabled and is set to receive focus, receives focus.
-- If the button has focus, Tab moves focus out of the button, to next focusable component in the tab order.
+- Tab moves focus onto the button.
+- If the button is disabled and not set to receive focus, Tab moves focus to the next focusable component.
+- If the button has focus, Tab moves focus to the next focusable component in the tabbing order.
```
Comment (origami-z) [inline site/docs/components/button/accessibility.mdx:37]: maybe move "tabbing" back to "tab" (see switch PR comment)

**Review by JoanaMMoreira — APPROVED** [review]

**Review by origami-z — APPROVED** [review]

**Pull request merged**